src/main/java/org/tikv/raw/RawKVClient.java

```diff
@@ -271,9 +264,11 @@ private void sendBatchPut(BackOffer backOffer, List<Batch> batches) {
       completionService.submit(
           () -> {
             BackOffer singleBatchBackOffer = ConcreteBackOffer.create(backOffer);
-            Pair<TiRegion, Metapb.Store> pair =
-                regionManager.getRegionStorePairByRegionId(batch.regionId);
-            RegionStoreClient client = RegionStoreClient.create(pair.first, pair.second, session);
+            RegionStoreClient client =
+                RegionStoreClient.create(
+                    batch.region,
+                    regionManager.getStoreById(batch.region.getContext().getPeer().getStoreId()),
+                    session);
```

Comment:
Why get store id from context? it should be accessible via getLeader().getStoreId()